tests/integration/mongodb/ldap/test_ldap.py

```
@@ -0,0 +1,230 @@
+
+    # Check that all units have restarted and updated their configuration.
+    for unit in ops_test.model.applications[db_app_name].units:
+        cat_cmd = cat_cmd.format(unit.name, path)
+        return_code, output, err = await ops_test.juju(*cat_cmd.split())
+
+        if return_code != 0:
+            raise ProcessError(f"Could not cat configuration. {output=} {err=}")
+
+        configuration = safe_load(output)
+        assert configuration["security"]["ldap"].get("userToDNMapping", "") != ""
+        assert configuration["security"]["ldap"]["authz"].get("queryTemplate", "") != ""
+
+    uri = await generate_mongodb_ldap_client(
+        ops_test,
+        substrate,
+        db_app_name,
+        database="superdb",
+        username="johndoe@superheroes",
+        password="dogood",
+    )
+    await execute_on_mongod(ops_test, db_app_name, substrate, uri, "db.test.insertOne({number: 2})")
+
+
+@pytest.mark.abort_on_fail
+async def test_remove_ldap_goes_to_blocked(ops_test: OpsTest, substrate: str):
+    """Only integrate ldap endpoint, should go into blocked state."""
+    db_app_name = await get_app_name(ops_test)
+
+    # We remove the first relation integrated, it should go into blocked state
+    await ops_test.model.applications[db_app_name].remove_relation(
+        f"{LDAP_OFFER}:ldap", f"{db_app_name}:ldap"
+    )
+
+    units = ops_test.model.applications[db_app_name].units
+
+    await ops_test.model.block_until(
+        *[lambda: unit.workload_status == "blocked" for unit in units], timeout=TIMEOUT
+    )
+
+    await wait_for_mongodb_units_blocked(
+        ops_test,
+        db_app_name,
+        status="GLauth TLS is integrated but LDAP is not.",
+        timeout=300,
+    )
+
+    # John should not be able to log in now.
+    uri = await generate_mongodb_ldap_client(
+        ops_test,
+        substrate,
+        db_app_name,
+        database="superdb",
+        username="johndoe@superheroes",
+        password="dogood",
+    )
+
```

for posterity, could you add a comment stating that we should expect the write to fail when the the ldap relation is missing?